src/components/auth/signIn/email/SignUpModalContent.tsx

```diff
@@ -0,0 +1,244 @@
+import HCaptcha from '@hcaptcha/react-hcaptcha'
+import { Keyring } from '@polkadot/api'
+import { stringToU8a, u8aToHex } from '@polkadot/util'
+import { isStr, toSubsocialAddress } from '@subsocial/utils'
+import { Button, Form, Input } from 'antd'
+import clsx from 'clsx'
+import jwtDecode from 'jwt-decode'
+import { RuleObject } from 'rc-field-form/lib/interface'
+import { useEffect, useRef, useState } from 'react'
+import { MutedDiv } from 'src/components/utils/MutedText'
+import {
+  emailSignUp,
+  getErrorMessage,
+  JwtPayload,
+  requestProof,
+  resendEmailConfirmation,
+} from 'src/components/utils/OffchainSigner/api/requests'
+import useMnemonicGenerate from 'src/components/utils/OffchainSigner/useMnemonicGenerate'
+import { hCaptchaSiteKey } from 'src/config/env'
+import useSignerExternalStorage from 'src/hooks/useSignerExternalStorage'
+import { StepsEnum, useAuth } from '../../AuthContext'
+import { EmailInput, PasswordInput } from './SignInModalContent'
+import styles from './SignInModalContent.module.sass'
+import useEncryptionToStorage from './useEncryptionToStorage'
+import { useFormValidation } from './useFormValidation'
+
+type FormValues = {
+  email: string
+  password: string
+  repeatPassword: string
+}
+
+type FieldName = keyof FormValues
+
+export const fieldName = (name: FieldName): FieldName => name
+
+type Props = {
+  setCurrentStep: (step: number) => void
+}
+
+const SignUpModalContent = ({ setCurrentStep }: Props) => {
+  const { mnemonic } = useMnemonicGenerate()
+  const { createEncryptedAccountAndSave } = useEncryptionToStorage()
+  const {
+    state: { email },
+    setPassword,
+    setEmail,
+  } = useAuth()
+  const { isValidEmail, isValidPassword } = useFormValidation()
+
+  const [form] = Form.useForm()
+
+  const [isFormValid, setIsFormValid] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const [token, setToken] = useState<string | undefined>()
+  const [captchaReady, setCaptchaReady] = useState(false)
+  const [loading, setLoading] = useState(false)
+
+  const hCaptchaRef = useRef(null)
+
+  const { setSignerTokensByAddress, setSignerTempRegisterAccount } = useSignerExternalStorage()
+
+  const onExpire = () => {
```

Comment:
this callbacks, along with onError, maybe its better to setError rather than just console.warn?
and if you change this, please change the sign in one too.

Reply:
Applied in 
[6b8698e](https://github.com/dappforce/polkaverse/pull/74/commits/6b8698e777e002dabb46284e32ce9c7fac3c46e9)